libs/ledger/src/chain/block_coordinator.cpp

```diff
@@ -549,21 +515,16 @@ BlockCoordinator::State BlockCoordinator::OnPreExecBlockValidation()
       return State::RESET;
     }
 
-    if (consensus_)
-    {
-      consensus_->UpdateCurrentBlock(*previous);  // Only update with valid blocks
-      auto result = consensus_->ValidBlock(*current_block_);
+    consensus_->UpdateCurrentBlock(*previous);  // Only update with valid blocks
+    auto result = consensus_->ValidBlock(*current_block_);
```

Comment:
Once @denis-trapeznikov stuff is done, the consensus valididity checks will not be in the block coordinator (instead guarantee main cahin provides valid blocks)